File: LeetCode/2269/main.cpp
class Solution {
public:
    int divisorSubstrings(int num, int k) {
        
        string s = to_string(num);
        int n = s.length();
        int ans = 0;
        
        for(int i = 0; i + k <= n; i++) {
            int d = stoi(s.substr(i, k));
            if(d != 0 && num % d == 0) {
                ans++;
            }
        }

        return ans;

    }
};

File: LeetCode/2269/main.py
class Solution:
    def divisorSubstrings(self, num: int, k: int) -> int:
        s = str(num)
        n = len(s)
        left, ans = 0, 0

        for i in range(n+1):

            if (i - left != k):
                continue

            c = s[left:i]

            if (int(c) != 0 and num % int(c) == 0):
                ans += 1

            left += 1

        return ans
